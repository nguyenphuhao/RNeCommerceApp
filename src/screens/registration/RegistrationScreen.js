import React, { useState } from 'react';
import { View, Platform, KeyboardAvoidingView } from 'react-native';
import { Input, Text, Button } from 'react-native-elements';
import Background from '../../assets/img/login-background.svg';
import styles from './styles';

const LoginScreen = () => {
  const [fullName, setFullName] = useState('');
  const [emailAddress, setEmailAddress] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  return (
    <>
      <Background width={'100%'} height={'100%'} style={styles.background} />
      <View style={styles.container}>
        <Text testId={'tstCaption'} style={styles.caption} h3>
          Ecommerce{'\n'}Store
        </Text>
        <KeyboardAvoidingView
          style={styles.form}
          behavior={Platform.Os === 'ios' ? 'padding' : 'height'}
        >
          <Input
            containerStyle={styles.input}
            testId={'tstFullName'}
            placeholder={'Full Name'}
            value={fullName}
            onChangeText={setFullName}
          />
          <Input
            containerStyle={styles.input}
            testId={'tstEmailAddress'}
            placeholder={'Email Address'}
            value={emailAddress}
            onChangeText={setEmailAddress}
          />
          <Input
            containerStyle={styles.input}
            testId={'tstPassword'}
            placeholder={'Password'}
            secureTextEntry={true}
            value={password}
            onChangeText={setPassword}
          />
          <Input
            containerStyle={styles.input}
            testId={'tstConfirmPassword'}
            placeholder={'Confirm Password'}
            secureTextEntry={true}
            value={confirmPassword}
            onChangeText={setConfirmPassword}
          />
        </KeyboardAvoidingView>
        <View style={styles.action}>
          <Button testId={'tstSignUp'} title="SIGN UP" />
          <Text testId={'tstSignIn'} style={styles.signInText}>
            Already have account? Sign in
          </Text>
        </View>
      </View>
    </>
  );
};

export default LoginScreen;
