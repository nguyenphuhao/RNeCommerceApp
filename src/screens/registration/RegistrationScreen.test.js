/**
 * @format
 */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import RegistrationScreen from './RegistrationScreen';

const shallowComponent = () => {
  const props = {};
  const wrapper = shallow(<RegistrationScreen {...props} />);
  return wrapper;
};

describe('RegistrationScreen exists components', () => {
  test('RegistrationScreen exists components - test snapshot', () => {
    const wrapper = shallowComponent();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
