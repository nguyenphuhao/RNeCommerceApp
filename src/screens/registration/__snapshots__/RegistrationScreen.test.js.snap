// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationScreen exists components RegistrationScreen exists components - test snapshot 1`] = `
<Fragment>
  <SvgMock
    height="100%"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    width="100%"
  />
  <View
    style={
      Object {
        "flex": 10,
        "padding": 10,
      }
    }
  >
    <Themed.Text
      h3={true}
      style={
        Object {
          "color": "#00a8f3",
          "elevation": 1,
          "flex": 3,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
      testId="tstCaption"
    >
      Ecommerce
      

      Store
    </Themed.Text>
    <KeyboardAvoidingView
      behavior="height"
      enabled={true}
      keyboardVerticalOffset={0}
      style={
        Object {
          "flex": 4,
          "justifyContent": "flex-end",
          "marginBottom": 20,
        }
      }
    >
      <ForwardRef(Themed.Input)
        containerStyle={Object {}}
        onChangeText={[Function]}
        placeholder="Full Name"
        testId="tstFullName"
        value=""
      />
      <ForwardRef(Themed.Input)
        containerStyle={Object {}}
        onChangeText={[Function]}
        placeholder="Email Address"
        testId="tstEmailAddress"
        value=""
      />
      <ForwardRef(Themed.Input)
        containerStyle={Object {}}
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        testId="tstPassword"
        value=""
      />
      <ForwardRef(Themed.Input)
        containerStyle={Object {}}
        onChangeText={[Function]}
        placeholder="Confirm Password"
        secureTextEntry={true}
        testId="tstConfirmPassword"
        value=""
      />
    </KeyboardAvoidingView>
    <View
      style={
        Object {
          "flex": 3,
          "paddingHorizontal": 10,
        }
      }
    >
      <ForwardRef(Themed.Button)
        testId="tstSignUp"
        title="SIGN UP"
      />
      <Themed.Text
        style={
          Object {
            "color": "#436bff",
            "fontSize": 18,
            "marginTop": 6,
            "textAlign": "center",
          }
        }
        testId="tstSignIn"
      >
        Already have account? Sign in
      </Themed.Text>
    </View>
  </View>
</Fragment>
`;
